### DISCUSSION TITLE
how to check if vehicle is personal vehicle

### DISCUSSION THREAD

**Discussion by lonelybud** [YimMenu/YimMenu#2816] [Q&A] [answered]
is there a native code that can help check if a vehicle is a personal vehicle  (yours or someone else doesn't matter) and not cloned/spawned randomly.

**Comment by FIying-Scotsman** [comment, marked as answer]
I'd recommend looking into vehicle decorators as there should be a flag for whether the vehicle is a player owned one or whether it's just something off the street.

From [this ](https://cookbook.fivem.net/2022/01/06/marking-a-vehicle-as-player-vehicle-for-game-code/)example:
`if DecorIsRegisteredAsType("Player_Vehicle", 3) then
    DecorSetInt(vehicle, "Player_Vehicle", -1)
end`

Other natives you'll likely need:

- [DECOR_EXIST_ON](https://nativedb.dotindustries.dev/gta5/natives/0x05661B80A8C9165F)
- [DECOR_REGISTER](https://nativedb.dotindustries.dev/gta5/natives/0x9FD90732F56403CE)

Keep in mind a lot of menus may set that decorator for menu-spawned vehicles which may throw whether it's a 'real' personal vehicle or just one created by a mod. This was part of a workaround to prevent the game kicking you from spawned vehicles (although just the "MPBitset" decorator was required - a lot of menus still set unused decorators for whatever reason). Other decorators exist but won't be useful for what you want to do (more for a [reference](https://gtaforums.com/topic/839159-decorator-namespace/?do=findComment&comment=1068416873) as there doesn't seem to be a complete version in 1 place).

The other method is to check for the personal vehicle blip ([DOES_BLIP_EXIST](https://nativedb.dotindustries.dev/gta5/natives/0xA6DB27D19ECBB7DA) - or do it using [classes](https://www.unknowncheats.me/forum/1296498-post315.html)) as spawned vehicles don't usually set a blip and passing the correct [blip ID](https://docs.fivem.net/docs/game-references/blips/) for what you want to look for.

**Reply by lonelybud** [reply]
thanks a lot friend. i will check and then tell if it works for me

**Reply by FIying-Scotsman** [reply]
@TheGreenBandit - Literally none of my post was AI generated. I'd be genuinely surprised if ChatGPT even knew how GTA worked (other than from leaked menu sources). You can also check the edits I made

**Reply by lonelybud** [reply]
decorator are helpful. Just checking these decorators whether there return values are `0` or not helped me identify PVs -

```C++
// should be a pv
DECORATOR::DECOR_GET_INT(veh, "MPBitset") || DECORATOR::DECOR_GET_INT(veh, "Player_Vehicle")
```
